src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -226,40 +226,26 @@ public FormValidation doTestHostname(@QueryParameter("hostname") final String ho
         }
     }
 
-    private boolean validateTargetApiURL(final String targetApiURL){
-        if(!DatadogClient.ClientType.HTTP.name().equals(reportWith)){
-            return true;
-        }
-        return StringUtils.isNotBlank(targetApiURL) && targetApiURL.contains("http");
-    }
-
     /**
      * @param targetApiURL - The API URL which the plugin will report to.
      * @return a FormValidation object used to display a message to the user on the configuration
      * screen.
      */
     public FormValidation doCheckTargetApiURL(@QueryParameter("targetApiURL") final String targetApiURL) {
-        if(!validateTargetApiURL(targetApiURL)) {
+        if(StringUtils.isNotBlank(targetApiURL) && !DatadogHttpClient.validateTargetURL(targetApiURL)) {
```

Comment:
I think we should separate concerns more clearly here:

- FormValidation ensures that the value in fields make sense and if not display a meaningful error message.
- DatadogClient checks connectivity to the provided endpoint

for example this function should only check that the string looks like a valid url, not that the url is a datadog one, nor that it is reachable.

Reply:
Right now validateTargetURL only checks if the URL contains "http", so I think it makes sense in GlobalConfiguration according to your points. Maybe it should  be taken out of  DatadogHttpClient?
```
 public static boolean validateTargetURL(String targetApiURL) {
        return targetApiURL.contains("http");
  }
```

Reply:
Moved!